examples/junit-example/src/test/java/io/github/checkleak/junitexample/AssertJTest.java

```diff
@@ -0,0 +1,32 @@
+package io.github.checkleak.junitexample;
+
+import io.github.checkleak.assertj.CheckLeakAssert;
+import io.github.checkleak.sample.SomeClass;
+import org.junit.jupiter.api.Test;
+
+public class AssertJTest {
+
+    @Test
+    void testAssertJ() throws Exception {
+
+        // I am keeping a reference live
+        SomeClass someObject = new SomeClass();
+
+        // I am starting the CheckLeak assertion API
+        CheckLeakAssert assertions = CheckLeakAssert.create();
+
+        // I'm checking if there are references. On this case I know I should have one object live, so I'm checking for 1
+        assertions.assertThatAllObjectsOf(SomeClass.class).hasSize(1);
```

Comment:
I thought about an assertion where you would have a report of holding references in case they went negative.


This is a valid assertion, sure..but it would be nice to throw a report in case it failed.


I thought about a replacement for this functionality when I opened this issue:

https://github.com/apache/activemq-artemis/blob/7fe1a20909e40a0b69db174e7340c740b3ebae49/tests/leak-tests/src/test/java/org/apache/activemq/artemis/tests/leak/MemoryAssertions.java#L51-L64

Reply:
you don't need to have the Wait clause of course... the Wait is something we wrote in Artemis.. (although it would be nice to have it part of AssertJ.